File: find-minimum-in-rotated-sorted-array/findMinimumInRotatedSortedArray.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        left = 0
        right = len(nums) - 1
        while left < right - 1:
            mid = left + (right - left) // 2
            if nums[mid] > nums[left]:
                if nums[mid] > nums[right]:
                    left = mid
                else:
                    right = mid
            else:
                if nums[mid] > nums[right]:
                    left = mid
                else:
                    right = mid
        if nums[left] < nums[right]:
            return nums[left]
        else:
            return nums[right]

File: find-minimum-in-rotated-sorted-array/findMinimumInRotatedSortedArray.java
public class Solution {
    public int findMin(int[] nums) {
        int left = 0, right = nums.length - 1;
        while(left + 1 < right) {
            int mid = left + (right - left) / 2;
            if(nums[mid] < nums[mid + 1] && nums[mid] < nums[mid - 1]){
                return nums[mid];
            }else if(nums[left] > nums[right] && nums[mid] > nums[left]) {
                left = mid;
            }else{
                right = mid;
            }
        }
        return nums[left] < nums[right] ? nums[left] : nums[right];
    }
}